### ISSUE TITLE
Idea: Use schema.org markup

### ISSUE THREAD

**Issue by olach** [moonstonemedia/Simple-Calendar#8]
Do you have plans to output markup of events according to [schema.org](http://schema.org)? The main benefit is that search engines understands the contents of the events and can display the events directly in the search results. The SEO ranking should also be improved.

Links:
[Documentation](http://schema.org/Event)
[Online schema creator](http://schema-creator.org/event.php)
[Google has some more information](https://developers.google.com/structured-data/rich-snippets/events)
[Structured Data Testing Tool](https://developers.google.com/structured-data/testing-tool/)

**Comment by nekojira** [comment]
Hi again olach, yes I also thought of this, I am familiar with schema.org and I agree with you

tldr: I'll try to put some of the meta tags for 3.0 release, but if I can't make it I'll definitely push this for the next iteration.

The current code that manages the calendar views output is not nice at all and it's the part I'm definitely less proud of. I brought it in a OOP context from the previous plugin version and should absolve the user needs but it still has to be improved, a lot. 

For the next iteration I would like to change a bit how the calendar views work, to make these classes a lot tidier and support theme templates. In order to do so I also need to separate some concerns. I have prepared a bit a collection object in includes/events/events.php (not being used/tested yet) that has fluent methods to get events from a collection (array) of event objects. The calendar, according to it's type (suppose, a monthly calendar) would query this collection 'get events for this month' and obtain an array with weeks/days to loop through and create a grid through template parts for example. Likewise this could help improving the js part, also not clean. And if there's a standard events query to loop through, that would also help designer making custom templates (I could throw a get_template_part() WordPress function here, with fallback on bundled template parts).

So while I'm fixing some markup/css in order to be usable, I'm not changing it a lot for now and would avoid to keep delaying the release too much to add new features.

**Comment by olach** [comment]
Thanks for your answer. Glad to hear that you have already thought of this. But as you say, maybe better to finish version 3 first, and have this as a nice 3.1 update. :)

**Comment by nekojira** [comment]
I added some tags, but this needs further work as I explained above

**Comment by olach** [comment]
Sweet, thanks!

**Comment by derekperkins** [comment]
:+1:
